pyomo/repn/plugins/gams_writer.py

```diff
@@ -78,6 +78,16 @@ def visit(self, node, values):
                 else:
                     tmp.append(val)
 
+        if node.__class__ is EXPR.LinearExpression:
+            argstr = []
+            for coef, var in zip(node.linear_coefs, node.linear_vars):
+                if var.fixed:
+                    argstr.append("{}".format(coef * var.value))
+                else:
+                    argstr.append("{} * {}".format(
+                        coef, var.to_string(verbose=False, smap=self.smap, compute_values=True)))
+            return " + ".join(argstr) + (" + {}".format(node.constant) if node.constant != 0 else "")
```

Comment:
We need to wrap formatting of constant values in `ftoa()` to better handle output precision.